File: flutter_sistem_rs/lib/models/dompet_medis_model.dart
// D:\Mobile App\flutter_sistem_rs\lib\models\dompet_medis_model.dart
class DompetMedis {
  final int idDeposit;
  final String noInvoice;
  final String namaPasien;
  final String nik;
  final String? namaBank;
  final double jumlahDeposit;
  final double saldoSisa;
  final String tanggalDeposit;
  final String? status;

  DompetMedis({
    required this.idDeposit,
    required this.noInvoice,
    required this.namaPasien,
    required this.nik,
    this.namaBank,
    required this.jumlahDeposit,
    required this.saldoSisa,
    required this.tanggalDeposit,
    this.status,
  });

  factory DompetMedis.fromJson(Map<String, dynamic> json) {
    return DompetMedis(
      idDeposit: json['IDDEPOSIT'] ?? 0,
      noInvoice: json['NOINVOICE'] ?? '-',
      namaPasien: json['NAMAPASIEN'] ?? '-',
      nik: json['NIK'] ?? '-',
      namaBank: json['NAMA_BANK'],
      jumlahDeposit: _parseToDouble(json['JUMLAH']),
      saldoSisa: _parseToDouble(json['SALDO_SISA']),
      tanggalDeposit: json['TANGGALDEPOSIT'] ?? '-',
      status: json['STATUS'],
    );
  }

  static double _parseToDouble(dynamic value) {
    if (value == null) return 0.0;
    if (value is double) return value;
    if (value is int) return value.toDouble();
    if (value is String) {
      return double.tryParse(value) ?? 0.0;
    }
    return 0.0;
  }
}

File: express_sistem_rs/src/models/dompetMedisModel.js
// D:\Mobile App\flutter_sistem_rs\express_sistem_rs\src\models\dompetMedisModel.js
import db from '../core/config/knex.js';

export const getDepositByNik = async (nik) => {
  return await db('deposit')
    .join('invoice', 'deposit.IDINVOICE', '=', 'invoice.IDINVOICE')
    .join('pasien', 'invoice.NIK', '=', 'pasien.NIK')
    .leftJoin('bank_account', 'deposit.IDBANK', '=', 'bank_account.IDBANK')
    .select(
      'deposit.IDDEPOSIT',
      'deposit.NODEPOSIT',
      'deposit.TANGGALDEPOSIT',
      'deposit.NOMINAL as JUMLAH',
      'deposit.STATUS',
      'invoice.NOINVOICE',
      'pasien.NIK',
      'pasien.NAMALENGKAP as NAMAPASIEN',
      'bank_account.NAMA_BANK'
    )
    .where('pasien.NIK', nik)
    .orderBy('deposit.TANGGALDEPOSIT', 'desc');
};

export const getPenggunaanByInvoice = async (noInvoice) => {
  return await db('deposit_penggunaan')
    .join('deposit', 'deposit_penggunaan.IDDEPOSIT', '=', 'deposit.IDDEPOSIT')
    .join('invoice', 'deposit_penggunaan.IDINVOICE', '=', 'invoice.IDINVOICE')
    .join('pasien', 'invoice.NIK', '=', 'pasien.NIK')
    .select(
      'deposit_penggunaan.IDPENGGUNAAN',
      'deposit_penggunaan.TANGGALPEMAKAIAN',
      'deposit_penggunaan.JUMLAH_PEMAKAIAN',
      'deposit.NODEPOSIT',
      'invoice.NOINVOICE',
      'pasien.NIK',
      'pasien.NAMALENGKAP as NAMAPASIEN'
    )
    
    .where('invoice.NOINVOICE', noInvoice)
    .orderBy('deposit_penggunaan.TANGGALPEMAKAIAN', 'desc');
};
